### PR TITLE
Build: align eslint config with 3.x branch as much as possible

### PR DESCRIPTION
**Pull request by timmywil**
### Summary ###

See https://github.com/jquery/jquery/pull/5522#discussion_r1695742620

### Checklist ###

* [x] New tests have been added to show the fix or feature works
* [x] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -298,6 +302,10 @@ export default [
 		rules: {
 			...jqueryConfig.rules,
 			"no-implicit-globals": "error",
+			"no-unused-vars": [
+				"error",
+				{ caughtErrorsIgnorePattern: "^_" }
```
Comment (mgol) [inline eslint.config.js:307]: Do we enable it even though it doesn't actually work? I thought ESLint fails on unrecognized options due to schema-based validation? 🤔

**Review by mgol — APPROVED** [review]
One comment but LGTM overall

File: eslint.config.js (hunk 1 above)
Comment (timmywil) [reply eslint.config.js:307]: I wondered that as well. Perhaps it validates rules, but not their options.

Comment (fasttime) [reply eslint.config.js:307]: ESLint does validate rule options, and `caughtErrorsIgnorePattern` is a valid option for the `no-unused-vars` in ESLint v.8.x: https://eslint.org/docs/v8.x/rules/no-unused-vars#caughterrorsignorepattern.

However, prior to v9.x (specifically to [this commit](https://github.com/eslint/eslint/commit/1a945890105d307541dcbff15f6438c19b476ade)), the option `caughtErrors` would default to `"none"`, preventing unused caught errors to be reported altogether. To get the same behavior as with ESLint v9.x, you may need to additionally set `caughtErrors: "all"` in the options.

Comment (timmywil) [reply eslint.config.js:307]: The point here is to align the eslint config as much as possible with the main branch. Adding another property would defeat the purpose. But, I don't mind that the rule won't be enforced until we can update to v9. That will also come with source changes elsewhere and they can be done with the update.

**Pull request merged**